### ISSUE TITLE
Incompatibility with Not Enough Ids, Makes certain Blocks Invisible

### ISSUE THREAD

**Issue by APolarBaer** [ChiselsAndBits/Chisels-and-Bits#181]
the last time i used this mod for 1.10.2 reasons seared brick would show up just fine but now for some reason its invisible whenever i chisel it D:

Edit: after finding one of my core mods was the actual cause i trimmed it down to just Not Enough IDs:
https://mods.curse.com/mc-mods/minecraft/235107-notenoughids

**Comment by AlgorithmX2** [comment]
I'm fairly certain that TiC White-lists seared brick themselves, Its normally a TileEntity I believe. You could try reporting it to them as well, maybe they changed something with how that works... or something else.

Otherwise I'll try and find some time to look into it.

**Comment by APolarBaer** [comment]
what do you mean?
it white lists ?

**Comment by AlgorithmX2** [comment]
I mean that normally C&B doesn't support that block, TiC sends C&B a message when its starting that overrides that behavior.

**Comment by APolarBaer** [comment]
it used to work just fine owo but okay

**Comment by AlgorithmX2** [comment]
They have been sending said message for a very long time, they are the reason that message exists int he first place. I'm just saying that its possible that the cause is on their end.

Maybe changing how their models work, perhaps referencing their TileEntity, and I don't support TileEntity's for good reason.

I don't know at this point, I've not had time to look into it.

**Comment by APolarBaer** [comment]
Botanias Metamorphic Mountain Stone Bricks do the same

**Comment by APolarBaer** [comment]
https://github.com/Vazkii/Botania/issues/2047

**Comment by AlgorithmX2** [comment]
Try an older version of C&B, specifically 12.1 if it works on the version, then its probobly an optimization causing issues.

**Comment by APolarBaer** [comment]
i will try tomorrow as im currently playing with my fiance ill post in the morning though, thanks for the suggestion. about 12 hours time

**Comment by APolarBaer** [comment]
seared stone works

**Comment by AlgorithmX2** [comment]
That was a fast 12 hours O_o

**Comment by AlgorithmX2** [comment]
If the issue doesn't appear in 12.1 then it is certainly a C&B issue, and I'll have to track it down.

**Comment by APolarBaer** [comment]
xD i  havent treid 12.1 i just thought id try a different tinkers block to see if it helped, sorry. but yeah ill let you know before 12 hours >w<

**Comment by AlgorithmX2** [comment]
Oh, I misunderstood its been a long day.

**Comment by APolarBaer** [comment]
Railcrafts Fitted Abyssal Stone also goes invisible
>W< every brick i try goes invisible rip Evilcrafts Dark Stone Bricks dont go invisible though

Blood Magics tiled obsidian path and Biome O Plentys Shale also disappear

**Comment by KnightMiner** [comment]
Other than having a TE (which is just data pointing towards the controller), seared bricks are just normal vanilla models, nothing has changed in our end recently.

Assuming brownstone works (which is identical as far as models), the issue is something with it having a TE.

**Comment by AlgorithmX2** [comment]
I'm starting to wonder what the connection is between all of these.. it can't be that they all have TE's I don't think this many blocks are white listed.

**Comment by AlgorithmX2** [comment]
I tried this TiC 1.10.2-2.5.6.447 I was unable to reproduce it, all versions of seared, cobble, bricks, small bricks, cracked and beyond all seem to work fine in my test environment.

Do you have any mods that alter rendering in anyway? Optifine or something else?

**Comment by APolarBaer** [comment]
https://minecraft.curseforge.com/projects/arctic-legion-server-pack my modpack, but I believe all I have is BetterFPS same as before when it worked

im using TConstruct 1.10.2-2.5.6b
going to try 12.1 right now for C&B

**Comment by APolarBaer** [comment]
12.1 works ... for the actual modpack

Making a custom profile matching my packs setup - most of the mods to try and pinpoint when it changed
my testing environment failed me because every version worked going to try putting 12.7 back on the server to see what happens...

nope, still does it. guess ill take the time to load up the whole pack to try and find it xD
12.7: Nope
12.6: Nope
12.5: Crashes on join server
12.4: Crashes on join server
12.3: havent been able to load a world for some reason
12.1: Works

**Comment by APolarBaer** [comment]
Ive loaded 12.1 on the server for now as its the only previous i can get to work the way it should. ive updated all the pending mods for my server but ive refrained from updating forge still as im having weird errors with it. 

it will be pack V 2.7 when its accepted

**Comment by AlgorithmX2** [comment]
Based on the fact that your "tests" resulted in it working everytime, to me it sounds like some other mod in your pack is causing the issue, I was also unable to reproduce this.

**Comment by APolarBaer** [comment]
any mod i can think it would be was included in the test environment though :/

**Comment by AlgorithmX2** [comment]
I don't have much to offer except for, Maybe its something that makes no sense.

**Comment by APolarBaer** [comment]
i guess ill just try updating later to see if i can get the new version of your mod sometime. if i find anything else ill let you know.

**Comment by AEnterprise** [comment]
@APolarBaer maybe try to reproduce with only C&B and tinkers?

**Comment by APolarBaer** [comment]
my test was 
Tinkers 
C&B 
Botania
Railcraft
Better FPS
Chunk Animator
Baubles
Evilcraft
Open Computers
Chisels and Bytes

and it worked jsut fine for 12.7
it just wasnt on a server. 
but on my modpack SSP does the same thing with 12.7 it disappears when i chisel it.

**Comment by APolarBaer** [comment]
it seems like one of my mods doesnt like a change that was made after 12.1 because 12.1 works just fine on the SMP server/client for my pack

**Comment by AEnterprise** [comment]
and now the question is: does it heapen with ONLY C&B and tinkers? you also mention betterFPS, i don't know that mod but it is an FPS mod thus is a very likely suspect of causing this issue, throw them all out to see if it still heapens, if it stops heapening throw them in one by one to find the culpit

**Comment by APolarBaer** [comment]
Just tinkers and C&B is fine for 12.7 just like in my testing with a few others including Better FPS

**Comment by APolarBaer** [comment]
i installed the mods i needed for my core mods on top of C&B and Tinkers and it happens again

**Comment by AlgorithmX2** [comment]
Well that makes sense then, since the optimizations in 12.2+ depend on MC's block count. Having a mod that alters that block count will break that optimization. And in turn cause things to fail.

**Comment by APolarBaer** [comment]
so the issue with Not Enough IDs should be fixed in the next update?

**Comment by AlgorithmX2** [comment]
First verify you don't have any issues with https://dvs1.progwml6.com/jenkins/job/Chisels-and-Bits/53/artifact/build/libs/chiselsandbits-12.0.53.jar

If that is good then I'll try and get the new build out later today.

**Comment by APolarBaer** [comment]
if its coming then no rush but ill check this out when i get a chance

Edit: works with NotEnoughIDs

thanks for the help and working with me (/being patient while i try and figure specifically what it was)

**Comment by danielhickman** [comment]
I've grown my mod pack a bit and after adding NotEnoughIDs, I have found a very similar issue.

I tried opening a freshly copied backup with one working set of mods then copied the backup again and tried with NotEnoughIDs (no other mods were installed during this test). The results were that most of my Chisel (not C&B) and Natura bits were invisible and didn't have a hitbox so couldn't be broken either. If stepped on, the player experienced the 'server-client desync disco glitch'.

Setup:
- Singleplayer
- Minecraft v1.12.2
- Forge v14.23.4.2707
- MCMultiPart v2.5.1
- C&B v14.17 (also tried v14.8)
- NotEnoughIDs v1.5.4.2
- And whole bunch of other mods that were ruled out

**Comment by AlgorithmX2** [comment]
@danielhickman I confirmed that the fix I committed back in 2016 is still working. Using FCB I Created far more blocks then the game without NotEnoughID's can handle, then installed NEID's and set piles of them down and chiseled them, no issues.

This test was done in SP, However as long as the id' mappings are synced properly when you log into the server there should be no difference here.

For further note, this issue manifested when you exceeded the normal ID limit, but from the sounds of your issue it seems like blocks that existed previously suddenly became corrupt after installing the mod.

I'm not sure where to go with that. This seems like a completely different issue, and i'm not sure that its C&B's fault. ID's shouldn't change, regardless if they are inside the vanilla range or in the expanded range that NotEnoughID's provide, the strange desync behavior your seeing suggests that the server and the client don't agree as well.

In summary, there are a large number of facts that don't really add up to any conclusive result.

**Comment by danielhickman** [comment]
Bummer. I don't know what's up either and it's fairly minor so I will probably just remove all of the blocks them all saving all of the patterns in my inventory.
